src/main/io/gps.c

```diff
@@ -449,197 +572,132 @@ void gpsInitUblox(void)
-                            tx_buffer.payload.gnss.configblocks[2].flags = UBLOX_GNSS_DEFAULT_SIGCFGMASK | 0x01000000; //last number is undocumented and was captured from uCenter
-                            if (gpsConfig()->gps_ublox_use_galileo) {
-                                tx_buffer.payload.gnss.configblocks[2].flags |= UBLOX_GNSS_ENABLE;
-                            }
-
-                            tx_buffer.payload.gnss.configblocks[3].gnssId = 3;    //BeiDou
-                            tx_buffer.payload.gnss.configblocks[3].resTrkCh = 8;  //min channels
-                            tx_buffer.payload.gnss.configblocks[3].maxTrkCh = 16;  //max channels
-                            tx_buffer.payload.gnss.configblocks[3].reserved1 = 0;
-                            tx_buffer.payload.gnss.configblocks[3].flags = UBLOX_GNSS_DEFAULT_SIGCFGMASK | 0x01000000; //last number is undocumented and was captured from uCenter
-
-                            tx_buffer.payload.gnss.configblocks[4].gnssId = 4;    //IMES
-                            tx_buffer.payload.gnss.configblocks[4].resTrkCh = 0;  //min channels
-                            tx_buffer.payload.gnss.configblocks[4].maxTrkCh = 8;  //max channels
-                            tx_buffer.payload.gnss.configblocks[4].reserved1 = 0;
-                            tx_buffer.payload.gnss.configblocks[4].flags = UBLOX_GNSS_DEFAULT_SIGCFGMASK | 0x03000000; //last number is undocumented and was captured from uCenter
-
-                            tx_buffer.payload.gnss.configblocks[5].gnssId = 5;    //QZSS
-                            tx_buffer.payload.gnss.configblocks[5].resTrkCh = 0;  //min channels
-                            tx_buffer.payload.gnss.configblocks[5].maxTrkCh = 3;  //max channels
-                            tx_buffer.payload.gnss.configblocks[5].reserved1 = 0;
-                            tx_buffer.payload.gnss.configblocks[5].flags = UBLOX_GNSS_DEFAULT_SIGCFGMASK | 0x05000000; //last number is undocumented and was captured from uCenter
-                            if (!gpsConfig()->gps_ublox_use_galileo) {
-                                tx_buffer.payload.gnss.configblocks[5].flags |= UBLOX_GNSS_ENABLE;
-                            }
-
-                            tx_buffer.payload.gnss.configblocks[6].gnssId = 6;    //GLONASS
-                            tx_buffer.payload.gnss.configblocks[6].resTrkCh = 8;  //min channels
-                            tx_buffer.payload.gnss.configblocks[6].maxTrkCh = 14; //max channels
-                            tx_buffer.payload.gnss.configblocks[6].reserved1 = 0;
-                            tx_buffer.payload.gnss.configblocks[6].flags = UBLOX_GNSS_ENABLE | UBLOX_GNSS_DEFAULT_SIGCFGMASK | 0x01000000; //last number is undocumented and was captured from uCenter
-                            
-                            ubloxSendConfigMessage((const uint8_t *) &tx_buffer, UBLOX_GNSS_MESSAGE_LENGTH);
+                    case 10:
```

Comment:
Is there not a nice UBLOX_ID for this?

Reply:
This switch case is a sequential state machine, and I'm reusing the state_position used in other places of the code as a numeric sub-state counter, so it's just reset to 0 in gpsSetState. I think the resulting code is more consistent and easier to read than before when messageState was used for tracking sub-states across different states.

So, naming each step in this particular switch case like GPS_CONFIGURE_STEP1, GPS_CONFIGURE_STEP2, etc. would not be consistent with how state_position is used elsewhere (just a numeric counter). It's very subtle and debatable, but IMHO for someone trying to understand the whole code it's more consistent as it is now.

Do you still think naming the steps would be better than just keeping the numbers?

Reply:
I'm OK with it. Thanks for explaining.